src/test/ui/suggestions/recover-invalid-float-invalid.rs

```diff
@@ -0,0 +1,21 @@
+fn main() {}
+
+fn a() {
```

Comment:
Not sure if there's a better way to test errors like this other than putting them in separate functions/items.

Reply:
They can all go in a single function.

Reply:
Unfortunately it seems having two of these errors in a single function means only the first will be tested, which is why I had them in separate functions. Just curious if there's a more idiomatic way to do this, but I'm not seeing anything. Not blocking.

Reply:
How are you writing the test? There are lots of examples in the test suite with more than one error per function, e.g. src/test/ui/attrs-resolution-errors.rs. Also see https://rustc-dev-guide.rust-lang.org/tests/ui.html for docs about writing UI tests.

Reply:
Something like the following fails prematurely:
```rust
fn main() {
    _ = .3u32;
    //~^ ERROR expected expression, found `.`
    _ = .0b0;
    //~^ ERROR expected expression, found `.`
    _ = .0o07;
    //~^ ERROR expected expression, found `.`
    _ = .0x0ABC;
    //~^ ERROR expected expression, found `.`
}
```

Reply:
Oh, I see, it's nothing to do with the test harness, it's the parser itself no longer able to recover. Hmm.

@compiler-errors: This PR is about some error suggestions relating to invalid literals. E.g. if you have `.3u32` the compiler current suggests:
```
error: float literals must have an integer part
 --> c.rs:2:9
  |
2 |     _ = .3u32;
  |         ^^^^^ help: must have an integer part: `0.3u32`
```
But that suggestion is bogus, because `0.3u32` isn't a valid literal:
```
error: invalid suffix `u32` for float literal
 --> c.rs:2:9
  |
2 |     _ = 0.3u32;
  |         ^^^^^^ invalid suffix `u32`
  |
  = help: valid suffixes are `f32` and `f64`
```
So I suggested that @cassaundra fix this, and this PR avoids the suggestion if the suffix isn't `f32`/`f64`, changing the error  to:
```
error: expected expression, found `.`
 --> c.rs:2:9
  |
2 |     _ = .3u32;
  |         ^ expected expression
```
This also prevents error recovery from happening, so any subsequent parse errors in the function aren't picked up.

Do you think this change is good enough as is? `.3u32` is weird enough that I'm not sure if trying to recover makes sense, but I'm no expert on error messages.

Reply:
@nnethercote yes, failing the parse and consuming the rest of the block is unfortunate but acceptable. We can file a ticket to follow up on this and attempt to recover by discarding only until the `;` (which I thought we did already in statements, but there might be some reason it doesn't trigger here). Splitting the test in one function per unparseable case should be enough to get around this issue in the tests.